### PR TITLE
using ossl_safe_getenv instead getenv

### LINKED ISSUE #20802: http_lib.c should not use getenv() directly
Hello,

Most of the OpenSSL codebase uses `ossl_safe_getenv`, except for `cpuid` which has its own simplified `ossl_getenv`. `http_lib.c` does not follow this in `use_proxy`:
https://github.com/openssl/openssl/blob/4e5f3d691343a691ddae739c51f7ae71e9893c98/crypto/http/http_lib.c#L256

nor in `OSSL_HTTP_adapt_proxy:
https://github.com/openssl/openssl/blob/4e5f3d691343a691ddae739c51f7ae71e9893c98/crypto/http/http_lib.c#L278

This is a portability issue potentially for platforms without a working `getenv` (the system call is no longer localized). I didn't dig further to see if not using the "safe" variant of `getenv` as implemented by OpenSSL could also have security impacts in the HTTP code.

Let me know if I missed something and my report does not make sense!
Best regards

### PR DESCRIPTION
**Pull request by bharsaklemukesh975**
Fixes #20802

### CODE REVIEW

**Comment by t8m** [comment]
The CI is relevant. The function you probably want to use is ossl_safe_getenv() not ossl_getenv().

**Review by t8m — APPROVED** [review]
This would be IMO acceptable with CLA: trivial. Either please amend the commit message (`git commit --amend`) to include `CLA: trivial` on a separate line, or please submit a real CLA. https://www.openssl.org/policies/cla.html

File: crypto/http/http_lib.c [hunk 1]
```diff
@@ -275,9 +275,9 @@ const char *OSSL_HTTP_adapt_proxy(const char *proxy, const char *no_proxy,
      * compatible with other HTTP client implementations like wget, curl and git
      */
     if (proxy == NULL)
-        proxy = getenv(use_ssl ? "https_proxy" : "http_proxy");
+        proxy = ossl_safe_getenv(use_ssl ? "https_proxy" : "http_proxy");
     if (proxy == NULL)
-        proxy = getenv(use_ssl ? OPENSSL_HTTP_PROXY :
+        proxy = ossl_safe_getenv(use_ssl ? OPENSSL_HTTP_PROXY :
                        OPENSSL_HTTPS_PROXY);
```
Comment (paulidale) [inline crypto/http/http_lib.c:281]: indentation nit

**Review by paulidale — APPROVED** [review]
Okay with tirival

**Comment by hlandau** [comment]
This PR is small enough that it would be reasonable to squash all commits into one commit with `CLA: trivial` in it. Every commit needs to have `CLA: trivial` in it before we can approve it.

**Comment by paulidale** [comment]
Squashing during merge is reasonable.

**Comment by t8m** [comment]
Squashed and merged to master, 3.1, and 3.0 branches. Thank you for your contribution.

**Pull request closed without merging**